src/test/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActorTest.scala

```diff
@@ -643,33 +631,32 @@ class TaskReplaceActorTest extends AkkaUnitTest with Eventually {
       eventually {
         queueOrder.verify(f.queue).add(_: AppDefinition, 1)
       }
-      assert(f.killService.numKilled == 0)
 
       val newInstanceId = Instance.Id.forRunSpec(newApp.id)
       val newTaskId = Task.Id(newInstanceId)
 
       //unhealthy
       ref ! InstanceHealthChanged(newInstanceId, newApp.version, newApp.id, healthy = Some(false))
       eventually {
-        f.killService.numKilled should be(0)
+        verify(f.tracker, never).setGoal(any, any, any)
       }
 
       //unready
       ref ! ReadinessCheckResult(ReadinessCheck.DefaultName, newTaskId, ready = false, None)
       eventually {
-        f.killService.numKilled should be(0)
+        verify(f.tracker, never).setGoal(any, any, any)
       }
 
       //healthy
       ref ! InstanceHealthChanged(newInstanceId, newApp.version, newApp.id, healthy = Some(true))
       eventually {
-        f.killService.numKilled should be(0)
+        verify(f.tracker, never).setGoal(any, any, any)
       }
 
       //ready
       ref ! ReadinessCheckResult(ReadinessCheck.DefaultName, newTaskId, ready = true, None)
       eventually {
-        f.killService.numKilled should be(1)
+        verify(f.tracker, once).setGoal(any, any, any)
```

Comment:
Yeah, I'd explicitly publish the event here instead of in an implementation of the mock.

Reply:
Second this; I'd prefer not to have mocks at all; I'd especially prefer not to have mocks with side-effects.

Reply:
Let's do the full implementation then. The `InstanceTrackActor` has the following props
```scala
  def props(
    metrics: ActorMetrics,
    taskLoader: InstancesLoader,
    updateStepProcessor: InstanceTrackerUpdateStepProcessor,
    stateOpResolver: InstanceUpdateOpResolver,
    repository: InstanceView,
    clock: Clock
```

`InstanceUpdateOpResolver` should be the actual implementation. The loader and view can be an in-memory implementation or mocks. Only the `InstanceTrackerUpdateStepProcessor` might be a little tricky. Well, when I think about it it could just use the event step and nothing else.

Reply:
I think we can just use `MarathonTestHelper.createTaskTracker`.